purchase_work_acceptance/models/purchase.py

```diff
@@ -109,3 +133,27 @@ def _prepare_account_move_line(self, move=False):
             res["quantity"] = wa_line.product_qty
             res["product_uom_id"] = wa_line.product_uom
         return res
+
+    @api.depends(
+        "wa_line_ids.wa_id.state",
+        "wa_line_ids.product_qty",
+        "product_qty",
+        "product_uom_qty",
+        "order_id.state",
+    )
+    def _compute_qty_accepted(self):
+        for line in self:
+            # compute qty_accepted
+            qty = 0.0
+            for wa_line in line.wa_line_ids:
+                if wa_line.wa_id.state in ["accept"]:
+                    qty += wa_line.product_uom._compute_quantity(
+                        wa_line.product_qty, line.product_uom
+                    )
+            line.qty_accepted = qty
+
+            # compute qty_to_accept
+            if line.order_id.state in ["purchase", "done"]:
+                line.qty_to_accept = line.product_uom_qty - line.qty_accepted
+            else:
+                line.qty_to_accept = 0
```

Comment:
Do we need to worry about state? Even if PO state is cancelled, I think qty_to_accept can still show a number. I think it will be more simple. 
```suggestion
            line.qty_to_accept = line.product_uom_qty - line.qty_accepted
```
From the code in view, the button will be hide because of state check anyway, right?
```
attrs="{'invisible': ['|', ('state', 'not in', ('purchase', 'done')), ('wa_accepted', '=', True)]}"
```

Reply:
fixed